src/Rules/FunctionDefinitionCheck.php

```diff
@@ -15,17 +15,21 @@
 class FunctionDefinitionCheck
 {
 
-	const VALID_TYPEHINTS = [
-		'self',
-		'static',
-		'array',
-		'callable',
-		'string',
-		'int',
-		'bool',
-		'float',
-		'void',
-		'iterable',
+	const COMMON_TYPEHINTS = [
+		'array' => 50100,
+		'callable' => 50400,
+		'string' => 70000,
+		'int' => 70000,
+		'bool' => 70000,
+		'float' => 70000,
+		'void' => 70100,
+		'iterable' => 70100,
+		'object' => 70200,
+	];
+
+	const METHOD_TYPEHINTS = [
```

Comment:
Not sure how much hipster that is, but you can do 

~~~php
const METHOD_TYPEHINTS = self::COMMON_TYPEHINTS + [
    ...
];
~~~